repo/common/common.go

```diff
@@ -2,26 +2,62 @@ package common
 
 import (
 	"fmt"
+	"regexp"
 	"strings"
+
+	"github.com/google/uuid"
 )
 
+// Find dynamic map key names passed  as Parent["foo"] notation
+var bracketsRe = regexp.MustCompile(`\[([^\[\]]*)\]`)
+
+// Normalization for supporting arbitrary dynamic keys with dots:
+// Gateway.PublicGateways["gw.example.com"].UseSubdomains
+// Pinning.RemoteServices["pins.example.org"].Policies.MFS.Enable
+func ConfigKeyToLookupData(key string) (normalizedKey string, dynamicKeys map[string]string) {
+	bracketedKeys := bracketsRe.FindAllString(key, -1)
+	dynamicKeys = make(map[string]string, len(bracketedKeys))
+	normalizedKey = key
+	for _, mapKeySegment := range bracketedKeys {
+		mapKey := strings.TrimPrefix(mapKeySegment, `["`)
+		mapKey = strings.TrimSuffix(mapKey, `"]`)
```

Comment:
This regex doesn't mandate the double quotes, so would match "[nft.storage]" for example and then the square brackets would fail to be trimmed out here because it's expecting double quotes.

Reply:
Hardened in: 36b62c1f8ff339ce489410c76da754d3cbc6c1f0